src/Discord.Net.Interactions/Attributes/ComplexParameterAttribute.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+
+namespace Discord.Interactions
+{
+    /// <summary>
+    ///     Registers a parameter as a complex parameter.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = true)]
+    public class ComplexParameterAttribute : Attribute
```

Comment:
It would make sense for the type to have the complex parameter attribute, ex:
```cs
[ComplexParameter]
public class Vector3
{
    ...
}
```

Reply:
Having the attribute in the method signature makes  reading commands much easier. Also the way you suggested makes it impossible to use types that were defined in external packages as complex parameters(like TimeSpan)

Reply:
How does the service pick what constructor to use when using external types like TimeSpan?

Reply:
Using [ComplexParameter(Type[])] overload. i know its ugly but its really useful. Im open to suggestions tho

Reply:
in C# 10 we can use attributes in lambdas, one idea is this
```cs
public async Task MyCommandAsync([ComplexParameter(([Summary("x")] x, [Summary("y")] y, [Summary("z")] z) => new Vector3(x,y,z))] Vector3 vect)
{
}
```

Or we can just pass in a func for building them like so
```cs
public Timespan TimespanBuilder([Summary("x")] unix)
{
    ...
}

public async Task MyCommandAsync([ComplexParameter(TimespanBuilder)] Timespan timespan)
{
}
```

Reply:
Wish they had released the generic attributes cuz without that using a lambda builder in the ctor of the attribute doesnt seem like a viable option. Builders might be the way to go but `MethodInfo` isnt a valid ctor parameter type for attributes so your second proposal isnt possible. We might need to create a TypeConverter/AutocompleteHandler type pattern to sort this out

Reply:
Wish they had released the generic attributes cuz without that using a lambda builder in the ctor of the attribute doesnt seem like a viable option. Builders might be the way to go but `MethodInfo` isnt a valid ctor parameter type for attributes so your second proposal isnt possible. We might need to create a TypeConverter/AutocompleteHandler type pattern to sort this out

Reply:
I think it makes more sense to just use type converters for external types and complex parameters for defined types in the assembly, adding a TypeConverter/AutocompleteHandler pattern just defeats the purpose

Reply:
Alright then, i will stick with [ComplexParameter(Type[])]. This should at least give users the option if they ever need it.